src/Server/MySQLHandler.cpp

```diff
@@ -371,6 +384,81 @@ void MySQLHandler::comQuery(ReadBuffer & payload)
     }
 }
 
+void MySQLHandler::comStmtPrepare(DB::ReadBuffer & payload)
+{
+    String statement;
+    readStringUntilEOF(statement, payload);
+    auto [is_success, statement_id] = emplacePreparedStatement(std::move(statement));
+    if (is_success)
+        packet_endpoint->sendPacket(PreparedStatementResponseOK(statement_id, 0, 0, 0), true);
+    else
+        packet_endpoint->sendPacket(ERRPacket(), true);
+}
+
+void MySQLHandler::comStmtExecute(ReadBuffer & payload)
+{
+    uint32_t statement_id;
+    payload.readStrict(reinterpret_cast<char *>(&statement_id), 4);
+
+    if (!prepared_statements.contains(statement_id))
+    {
+        LOG_ERROR(log, "Could not find prepared statement with id {}", statement_id);
+        packet_endpoint->sendPacket(ERRPacket(), true);
+        return;
+    }
+
+    // Temporary workaround as we work only with queries that do not bind any parameters atm
+    ReadBufferFromString com_query_payload(prepared_statements.at(statement_id));
+    MySQLHandler::comQuery(com_query_payload, true);
+};
+
+void MySQLHandler::comStmtClose(ReadBuffer & payload)
+{
+    uint32_t statement_id;
+    payload.readStrict(reinterpret_cast<char *>(&statement_id), 4);
+    erasePreparedStatement(statement_id);
+
+    // https://dev.mysql.com/doc/dev/mysql-server/latest/page_protocol_com_stmt_close.html
+    // No response packet is sent back to the client.
+};
+
+MySQLHandler::EmplacePreparedStatementResult MySQLHandler::emplacePreparedStatement(String statement)
+{
+    static constexpr size_t MAX_PREPARED_STATEMENTS = 10'000;
+    std::lock_guard<std::mutex> lock(prepared_statements_mutex);
+    if (prepared_statements.size() > MAX_PREPARED_STATEMENTS) /// Shouldn't happen in reality as COM_STMT_CLOSE cleans up the elements
+    {
+        LOG_ERROR(log, "Too many prepared statements");
+        current_prepared_statement_id = 0;
+        prepared_statements.clear();
+        return std::make_pair(false, 0);
+    }
+    uint32_t statement_id = current_prepared_statement_id++;
+
+    // Key collisions should not happen here, as we remove the elements from the map with COM_STMT_CLOSE,
+    // and we have quite a big range of available identifiers with 32-bit unsigned integer
+    if (prepared_statements.contains(statement_id))
+    {
+        LOG_ERROR(
+            log,
+            "Failed to store a new statement `{}` with id {}; it is already taken by `{}`",
+            statement,
+            statement_id,
+            prepared_statements.at(statement_id));
+        return std::make_pair(false, 0);
+    }
+
+    prepared_statements.emplace(statement_id, statement);
+    return std::make_pair(true, statement_id);
```

Comment:
`return {true, statement_id};` (semantic sugar)

(similar syntax can be used above)